posixlib/src/main/resources/scala-native/time.c

```diff
@@ -40,6 +46,8 @@ static void tm_init(struct tm *tm, struct scalanative_tm *scala_tm) {
     tm->tm_wday = scala_tm->tm_wday;
     tm->tm_yday = scala_tm->tm_yday;
     tm->tm_isdst = scala_tm->tm_isdst;
+    tm->tm_gmtoff = scala_tm->tm_gmtoff;
```

Comment:
I believe that the OS tm fields can not be blindly set.  The code may be 
operating on an OS which uses the "short" 36 byte POSIX basic tm.
I think code similar to that below is needed;
```
if (sizeof(struct tm) >=  sizeof(struct scalanative_tm)) {
    tm->tm_gmtoff = scala_tm->tm_gmtoff;
    tm->tm_zone = scala_tm->tm_zone;
}
```

Reply:
@LeeTibbert it won't compile if `struct tm` hasn't got this fields.

I cant wrap it with `#ifdef` and setup `0` and `NULL` to new fields, but I'm not sure that it is worth special of we can enable it almost everywhere with magic `#define`

Reply:
On short structures, the non-existant fields of the structure will not get set (because of the size check).

Yes, an #ifdef is better than a runtime check. That is, setting tm_gmtoff and tm_zone here is
done only on systems which _have_ those fields.  

Since the basic POSIX specification is "short", 36-byte, I think we can not assume the longer 
structure everywhere.

Reply:
@LeeTibbert right now it is compile time check. It won't compile on system which hasn't got supported longer structure.

As result no magic debugging with zero value.

I guess it is the best behaviour, isn't it?